apis/vshn/v1/dbaas_vshn_redis.go

```diff
@@ -0,0 +1,88 @@
+package v1
+
+import (
+	v1 "github.com/vshn/component-appcat/apis/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Workaround to make nested defaulting work.
+// kubebuilder is unable to set a {} default
+//go:generate yq -i e ../../generated/vshn.appcat.vshn.io_vshnredis.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.default={})"
+//go:generate yq -i e ../../generated/vshn.appcat.vshn.io_vshnredis.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.size.default={})"
+//go:generate yq -i e ../../generated/vshn.appcat.vshn.io_vshnredis.yaml --expression "with(.spec.versions[]; .schema.openAPIV3Schema.properties.spec.properties.parameters.properties.service.default={})"
+
+// +kubebuilder:object:root=true
+
+// VSHNRedis is the API for creating Redis clusters.
+type VSHNRedis struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of a VSHNRedis.
+	Spec VSHNRedisSpec `json:"spec"`
+
+	// Status reflects the observed state of a VSHNRedis.
+	Status VSHNRedisStatus `json:"status,omitempty"`
+}
+
+// VSHNRedisSpec defines the desired state of a VSHNRedis.
+type VSHNRedisSpec struct {
+	// Parameters are the configurable fields of a VSHNRedis.
+	Parameters VSHNRedisParameters `json:"parameters,omitempty"`
+}
+
+// VSHNRedisParameters are the configurable fields of a VSHNRedis.
+type VSHNRedisParameters struct {
+	// Service contains Redis DBaaS specific properties
+	Service VSHNRedisServiceSpec `json:"service,omitempty"`
+
+	// Size contains settings to control the sizing of a service.
+	Size VSHNRedisSizeSpec `json:"size,omitempty"`
+}
+
+// VSHNRedisServiceSpec contains Redis DBaaS specific properties
+type VSHNRedisServiceSpec struct {
+	// +kubebuilder:validation:Enum="6.2";"7.0"
+	// +kubebuilder:default="7.0"
+
+	// Version contains supported version of Redis.
+	// Multiple versions are supported. The latest version "7.0" is the default version.
+	Version string `json:"version,omitempty"`
```

Comment:
If I get this right this version value is used directly as the image tag. So we'll always update patch versions on restart.

I think this is fine for the first iteration, but we'll need to think about how we handle this in the future, I'd rather have proper version pins here and not update it whenever.

Reply:
Yes correct. We will have to think how to handle this later on